txstratum/cli.py

```diff
@@ -51,110 +53,154 @@ def create_parser() -> ArgumentParser:
+        tx_filters: List[TXFilter] = []
+        toiclient: Optional[TOIAsyncClient] = None
+
+        if args.prometheus:
+            from txstratum.prometheus import PrometheusExporter
+            metrics = PrometheusExporter(manager, pubsub, args.prometheus)
+            metrics.start()
+
+        if args.prometheus_port:
+            from txstratum.prometheus import HttpPrometheusExporter
+            http_metrics = HttpPrometheusExporter(manager, pubsub, args.prometheus_port)
+            http_metrics.start()
+            logger.info('Prometheus metrics server running at 0.0.0.0:{}...'.format(args.prometheus_port))
+
+        if args.ban_addrs or args.ban_tx_ids:
+            tx_filters.append(FileFilter.load_from_files(args.ban_tx_ids, args.ban_addrs))
+
+        if args.toi_url or args.toi_apikey:
+            if not (args.toi_url and args.toi_apikey):
+                raise ValueError("Should pass both toi_url and toi_apikey")
+            toiclient = TOIAsyncClient(args.toi_url, args.toi_apikey)
+            tx_filters.append(TOIFilter(toiclient, block=args.toi_fail_block))
+
+        api_app = App(manager, max_tx_weight=args.max_tx_weight, max_timestamp_delta=args.max_timestamp_delta,
+                      tx_timeout=args.tx_timeout, fix_invalid_timestamp=args.fix_invalid_timestamp,
+                      only_standard_script=not args.allow_non_standard_script, tx_filters=tx_filters)
+        logger.info('API Configuration', max_tx_weight=api_app.max_tx_weight, tx_timeout=api_app.tx_timeout,
+                    max_timestamp_delta=api_app.max_timestamp_delta,
+                    fix_invalid_timestamp=api_app.fix_invalid_timestamp,
+                    only_standard_script=api_app.only_standard_script,
+                    tx_filters=tx_filters)
+
+        web_runner = web.AppRunner(api_app.app, logger=logger)
+        loop.run_until_complete(web_runner.setup())
+        site = web.TCPSite(web_runner, '0.0.0.0', args.api_port)
+        loop.run_until_complete(site.start())
+
+        self.register_signal_handlers()
+
+        try:
+            logger.info('Stratum Server running at 0.0.0.0:{}...'.format(args.stratum_port))
+            logger.info('TxMining API running at 0.0.0.0:{}...'.format(args.api_port))
+            if args.testnet:
+                logger.info('Running with testnet config file')
+            loop.run_forever()
+        except KeyboardInterrupt:
+            pass
 
-    for tx_filter in tx_filters:
-        loop.run_until_complete(tx_filter.close())
-    server.close()
-    loop.run_until_complete(server.wait_closed())
-    loop.run_until_complete(backend.stop())
-    loop.close()
+        logger.info('Stopping...')
+        for tx_filter in tx_filters:
+            loop.run_until_complete(tx_filter.close())
+        server.close()
+        loop.run_until_complete(server.wait_closed())
+        loop.run_until_complete(backend.stop())
+        loop.run_until_complete(manager.stop())
```

Comment:
We were not stopping the manager before. Was it a bug?

Reply:
Yeah, I think so. But it didn't generate bad consequences, just a warning from asyncio that a task was still running (which is the task that the manager.stop method stops)